### PR TITLE
Reference mismatches fixed, SCA

### PR DESCRIPTION
**Pull request by kalessil**
SCA with new version of Php Inspections (EA Extended) included

### CODE REVIEW

**Comment by kalessil** [comment]
@keradus : reference mismatches resolved

**Comment by keradus** [comment]
cc @stof

**Comment by kalessil** [comment]
@stof : regarding references mismatches, is everything legal <a href="\Symfony\Component\PropertyAccess\PropertyAccessor">here</a> (Symfony)?

**Comment by GrahamCampbell** [comment]
Any news on this?

File: Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php [hunk 1]
```diff
@@ -139,7 +139,7 @@ private function isValidControl(Tokens $tokens, Token $docsToken, $controlIndex)
         $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
         $docsContent = $docsToken->getContent();
 
-        for ($index = $index + 1; $index < $endIndex; ++$index) {
+        for (++$index; $index < $endIndex; ++$index) {
```
Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php]: You are actually making the code slower with this change.

File: Symfony/CS/Tests/Tokenizer/TokensTest.php [hunk 2]
```diff
@@ -817,7 +817,7 @@ public function testShortOpenTagMonolithicPhpDetection($source, $monolitic)
          * short_open_tag setting is ignored by HHVM
          * @see https://github.com/facebook/hhvm/issues/4758
          */
-        if (!ini_get('short_open_tag') && !defined('HHVM_VERSION')) {
+        if (!defined('HHVM_VERSION') && !ini_get('short_open_tag')) {
```
Comment (GrahamCampbell) [inline Symfony/CS/Tests/Tokenizer/TokensTest.php]: Why are these changed? This doesn't make sense. You are making average execution slower because HHVM_VERSION is always not defined during most executions.

File: Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php (hunk 1 above)
Comment (kalessil) [inline Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php]: How, even opCache  uses prefixed increments as an optimization strategy?

**Comment by kalessil** [comment]
I reverted ref-mismatches here, so just SCA needs a review.

File: Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php (hunk 1 above)
Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php]: To optimise postincrementation, yes.

Comment (kalessil) [inline Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php]: `$index = $index + 1` is 3 operations:
- read
- sum
- strore

When `++$index` is one.

File: Symfony/CS/Tests/Tokenizer/TokensTest.php (hunk 2 above)
Comment (kalessil) [inline Symfony/CS/Tests/Tokenizer/TokensTest.php]: Possible. @keradus : can you give here your resolution please?

Comment (keradus) [inline Symfony/CS/Tests/Tokenizer/TokensTest.php]: agreed with @GrahamCampbell

File: Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php]: One way or another, less readable for me.
- set initial value of iterator
- test iterator
- change iterator

now the first step is changing too

Comment (kalessil) [inline Symfony/CS/Fixer/Symfony/PhpdocToCommentFixer.php]: `less readable for me` - this good reason to roll back

File: Symfony/CS/Tests/Tokenizer/TokensTest.php (hunk 2 above)
Comment (kalessil) [inline Symfony/CS/Tests/Tokenizer/TokensTest.php]: @GrahamCampbell : you was right, will roll back this.

File: Symfony/CS/Fixer/PSR2/EofEndingFixer.php [hunk 3]
```diff
@@ -27,7 +27,7 @@ public function fix(\SplFileInfo $file, $content)
 
         $content = rtrim($content);
 
-        if (strlen($content)) {
+        if ('' !== $content) {
```
Comment (keradus) [inline Symfony/CS/Fixer/PSR2/EofEndingFixer.php]: this could be nice new fixer, same like `is_null` to `null ===`)

File: Symfony/CS/Fixer/Symfony/PhpdocIndentFixer.php [hunk 4]
```diff
@@ -40,8 +40,8 @@ public function fix(\SplFileInfo $file, $content)
 
             // ignore inline docblocks
             if (
+                $prevToken->equalsAny(array(';', '{')) ||
                 ($prevToken->isWhitespace(array('whitespaces' => " \t")) && !$tokens[$index - 2]->isGivenKind(T_OPEN_TAG))
-                || $prevToken->equalsAny(array(';', '{'))
```
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocIndentFixer.php]: why is that change ?

Comment (kalessil) [inline Symfony/CS/Fixer/Symfony/PhpdocIndentFixer.php]: It was proposal from analyser (costs estimation), as I understood semantics remains unchanged.
I'll revert if there are any doubts.

File: Symfony/CS/Fixer/PSR2/EofEndingFixer.php (hunk 3 above)
Comment (kalessil) [inline Symfony/CS/Fixer/PSR2/EofEndingFixer.php]: It's not so simple - there are several different patterns + yoda/no-yoda style doubles variants.
I can share all patterns from my analyser if you'll create a ticket for this.

File: Symfony/CS/Fixer/Symfony/PhpdocIndentFixer.php (hunk 4 above)
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocIndentFixer.php]: And why it should be faster?

Comment (kalessil) [inline Symfony/CS/Fixer/Symfony/PhpdocIndentFixer.php]: (method-call && !method-call) || (method-call) - worth case 3 calls, best case 2 calls
(method-call) || (method-call && !method-call) - worth case 3 calls, best case 1 call

But that's partially lie - distribution inline/general doc-block in real projects can break this assumption easily.

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/PhpdocIndentFixer.php]: exactly. and that new first method-call (with whitespace) uses regexp, so...

Comment (kalessil) [inline Symfony/CS/Fixer/Symfony/PhpdocIndentFixer.php]: Will revert then.

PS: In 1.7 Token operates internally on plain string API as I can see.

**Comment by keradus** [comment]
Thank you @kalessil.

**Comment by kalessil** [comment]
Yoohoo, thank you @keradus
